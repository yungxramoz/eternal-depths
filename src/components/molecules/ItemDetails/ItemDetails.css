.item-container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: row;
  column-gap: 1.5em;
  width: 100%;
  min-height: 80px;
}

.item-icon {
  flex-grow: 0;
  flex-basis: 44px;
  flex-shrink: 0;
}

.item-content {
  flex-grow: 1;
  text-align: left;
}

.item-title {
  margin: 0;
}

.item-subtitle {
  margin: 0 0 0.5em 0;
  font-size: x-small !important;
}

.item-description {
  margin: 0;
  margin-top: 4px;
  font-size: xx-small !important;
  line-height: 18px !important;
}
