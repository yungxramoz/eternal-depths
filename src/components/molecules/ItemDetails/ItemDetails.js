import React from 'react'
import './ItemDetails.css'
import RpgIcon from '../../atoms/RpgIcon/RpgIcon'

const ItemDetails = ({ title, subtitle, description, icon }) => {
  return (
    <div className="item-container">
      {icon && <RpgIcon className="item-icon" icon={icon} />}
      <div className="item-content">
        <p className="item-title">{title}</p>
        <p className="item-subtitle">
          <i>{subtitle}</i>
        </p>
        <p className="item-description">{description}</p>
      </div>
    </div>
  )
}

export default ItemDetails
