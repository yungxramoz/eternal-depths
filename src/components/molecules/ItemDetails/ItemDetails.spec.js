import { render, screen } from '@testing-library/react'
import React from 'react'
import ItemDetails from './ItemDetails'

describe('ItemDetails', () => {
  it('renders correctly', () => {
    render(
      <ItemDetails
        title="test-title"
        subtitle="test-subtitle"
        description="test-description"
        icon="test-icon"
      />,
    )
    expect(document.body).toMatchSnapshot('ItemDetails')
  })

  it('does not render RpgIcon if icon prop is not provided', () => {
    render(
      <ItemDetails
        title="test-title"
        subtitle="test-subtitle"
        description="test-description"
      />,
    )
    expect(screen.queryByTestId('rpg-icon')).toBeNull()
  })
})
