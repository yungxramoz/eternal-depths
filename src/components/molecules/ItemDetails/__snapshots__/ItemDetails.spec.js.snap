// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetails renders correctly: ItemDetails 1`] = `
<body>
  <div>
    <div
      class="item-container"
    >
      <div
        class="rpgui-icon test-icon item-icon"
      />
      <div
        class="item-content"
      >
        <p
          class="item-title"
        >
          test-title
        </p>
        <p
          class="item-subtitle"
        >
          <i>
            test-subtitle
          </i>
        </p>
        <p
          class="item-description"
        >
          test-description
        </p>
      </div>
    </div>
  </div>
</body>
`;
